src/app/page/Signup/SignupScene.jsx

```diff
@@ -46,59 +46,54 @@ function SignupScene(props) {
     };
   }
 
-  function handleSignupSuccess(user) {
-    setValues({
-      ...values,
+  async function handleSignupSuccess(user) {
+    let data = {
+      ...getPayload(),
       phoneNumber: user.phoneNumber,
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
-    });
-    updateUser();
-    setActiveTab(Tabs.CONNECT);
-    return false;
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.CONNECT));
   }
-  function handleConnectSuccess(user) {
-    setValues({
-      ...values,
+
+  async function handleConnectSuccess(user) {
+    let data = {
+      ...getPayload(),
       displayName: user.displayName,
       email: user.email,
       photoURL: user.photoURL,
+      uid: user.uid,
+    };
+    await updateUser(data, () => setActiveTab(Tabs.PROFILE));
+  }
+
+  async function handleProfileUpdate() {
+    await updateUser({ ...getPayload() }, () => {
+      setActiveTab(Tabs.SUCCESS);
+      snackbarContext.show({
+        message: "Congratulations, your volunteer account has been created!",
+        type: "success",
+      });
```

Comment:
I feel that some of the nested await function calls have gotten a little confusing.

I think we can consolidate some of these handlers to remove some of the jumping around of the logic to something like this (not exact)

```
// Update profile function gets invoked
async function handleProfileUpdate(payload) {
  try { // Everything should only happen upon a successful user.update
    await User.update(payload.uid, payload); // This promise should be resolved before this logic can continue. If there is an error thrown here, caught by catch()
    setValues(payload); // setValues after user is updated
	setActiveTab(Tabs.SUCCESS);
    snacbarContext.show({ // snackbar show success after everything above has run
      message: "Success", type: "success"
    })
  } catch (error) { // show error if there is an error updating user.
    snacbarContext.show({
      snackbarContext.show({message:"Error", type:"error"})
    })
  }
}

```

I think something like this would make more sense rather than passing around and invoking the `callback` param that we're passing into `updateUser`.

Reply:
These all make sense, I'll work on it tonight!